components/org.wso2.carbon.identity.oauth.endpoint/src/test/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpointTest.java

```diff
@@ -389,6 +389,85 @@ public void testTokenErrorResponse(String errorCode, Object headerObj, int expec
         assertTrue(response.getEntity().toString().contains(expectedErrorCode), "Expected error code not found");
     }
 
+    @DataProvider(name = "testTokenErrorResponseWithCustomParametersDataProvider")
+    public Object[][] testTokenErrorResponseWithCustomParametersDataProvider() {
+
+        ResponseHeader contentType = new ResponseHeader();
+        contentType.setKey(OAuth.HeaderType.CONTENT_TYPE);
+        contentType.setValue(OAuth.ContentType.URL_ENCODED);
+
+        ResponseHeader[] headers = new ResponseHeader[]{contentType};
+
+        Map<String, String> customErrorsMap = new HashMap<>();
+        customErrorsMap.put("customMessage1", "This is custom message 1");
+        customErrorsMap.put("customMessage2", "This is custom message 2");
+
+        return new Object[][]{
+                {TOKEN_ERROR, headers, customErrorsMap, HttpServletResponse.SC_BAD_REQUEST, TOKEN_ERROR}
```

Comment:
We can add some other combinations with empty error map, empty headers to the data provider

Reply:
Fixed.